test/osb_test/osb_test.go

```diff
@@ -728,6 +728,29 @@ var _ = Describe("Service Manager OSB API", func() {
 		})
 	})
 
+	Describe("Malfunctioning broker", func() {
+		Context("when broker response is not a valid json", func() {
+			It("should return an OSB compliant error", func() {
+				failingBrokerServer.ServiceInstanceHandler = func(rw http.ResponseWriter, _ *http.Request) {
+					rw.WriteHeader(http.StatusCreated)
+					rw.Write([]byte("[not a json]"))
+				}
+				ctx.SMWithBasic.PUT(smUrlToFailingBroker+"/v2/service_instances/12345").WithHeader("X-Broker-API-Version", "oidc_authn.13").
+					WithJSON(getDummyService()).Expect().Status(http.StatusCreated).JSON().Object().Value("description").String().Match("Broker with name .* response is not a valid json: \\[not a json\\]")
+			})
+		})
+
+		Context("when broker response error description is empty but", func() {
+			It("should assing default description", func() {
+				failingBrokerServer.ServiceInstanceHandler = func(rw http.ResponseWriter, _ *http.Request) {
+					rw.WriteHeader(http.StatusBadRequest)
+					rw.Write([]byte(`{"error": "ErrorType"}`))
+				}
+				ctx.SMWithBasic.PUT(smUrlToFailingBroker+"/v2/service_instances/12345").WithHeader("X-Broker-API-Version", "oidc_authn.13").
+					WithJSON(getDummyService()).Expect().Status(http.StatusBadRequest).JSON().Object().Value("description").String().Match("Broker with name .* failed with: {}")
+			})
+		})
+	})
```

Comment:
Additional test cases:
* Broker returns a JSON literal, e.g. `3`, `"abc"`, `true`, `null`
* Broker returns a JSON array, e.g. `["a","b","c"]`

In all these cases we should include the original response in the description